### PR TITLE
session: fix data race in the TestUpgradeVersion83

### LINKED ISSUE #31831: DATA RACE at the Domain.bindHandle
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
==================
WARNING: DATA RACE
Write at 0x00c000f643e0 by goroutine 174:
  github.com/pingcap/tidb/domain.(*Domain).LoadBindInfoLoop()
      /go/tidb/domain/domain.go:1076 +0xe4
  github.com/pingcap/tidb/session.BootstrapSession()
      /go/tidb/session/session.go:2655 +0x5aa
  github.com/pingcap/tidb/session.TestUpgradeVersion83()
      /go/tidb/session/bootstrap_test.go:832 +0x104
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
Previous read at 0x00c000f643e0 by goroutine 55:
  github.com/pingcap/tidb/domain.(*Domain).BindHandle()
      /go/tidb/domain/domain.go:1068 +0x22e
  github.com/pingcap/tidb/planner.getBindRecord()
      /go/tidb/planner/optimize.go:458 +0x21f
  github.com/pingcap/tidb/planner.matchSQLBinding()
      /go/tidb/planner/optimize.go:90 +0x8f
  github.com/pingcap/tidb/planner.Optimize()
      /go/tidb/planner/optimize.go:147 +0x631
  github.com/pingcap/tidb/executor.(*Compiler).Compile()
      /go/tidb/executor/compiler.go:77 +0x704
  github.com/pingcap/tidb/session.(*session).ExecuteStmt()
      /go/tidb/session/session.go:1696 +0x757
  github.com/pingcap/tidb/session.(*session).ExecuteInternal()
      /go/tidb/session/session.go:1429 +0x4ce
  github.com/pingcap/tidb/statistics/handle.(*Handle).initStatsHistograms()
      /go/tidb/statistics/handle/bootstrap.go:161 +0xe9
  github.com/pingcap/tidb/statistics/handle.(*Handle).InitStats()
      /go/tidb/statistics/handle/bootstrap.go:402 +0x1f0
  github.com/pingcap/tidb/domain.(*Domain).loadStatsWorker()
      /go/tidb/domain/domain.go:1341 +0x1f8
  github.com/pingcap/tidb/domain.(*Domain).UpdateTableStatsLoop·dwrap·18()
      /go/tidb/domain/domain.go:1281 +0x39
Goroutine 174 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1306 +0x726
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1598 +0x99
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1596 +0x7ca
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1504 +0x9d1
  github.com/pingcap/tidb/testkit/testmain.(*testingM).Run()
      /go/tidb/testkit/testmain/wrapper.go:28 +0x42
  go.uber.org/goleak.VerifyTestMain()
      /home/prow/go/pkg/mod/go.uber.org/goleak@v1.1.12/testmain.go:53 +0x59
  github.com/pingcap/tidb/session.TestMain()
      /go/tidb/session/main_test.go:74 +0xad3
  main.main()
      _testmain.go:205 +0x264
Goroutine 55 (running) created at:
  github.com/pingcap/tidb/domain.(*Domain).UpdateTableStatsLoop()
      /go/tidb/domain/domain.go:1281 +0x1f8
  github.com/pingcap/tidb/session.BootstrapSession()
      /go/tidb/session/session.go:2708 +0x804
  github.com/pingcap/tidb/session.createStoreAndBootstrap()
      /go/tidb/session/main_test.go:89 +0x64
  github.com/pingcap/tidb/session.TestUpgradeVersion83()
      /go/tidb/session/bootstrap_test.go:829 +0x66
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
==================
```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #31831

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tisonkun — APPROVED** [review]
Approve.

But I'm afraid that if you fix it partially, you may fight with CI for the failures caused by files you changed exactly.

If you're not in a hurry, I suggest you move related tests in bootstrap_test.go to upgrade_test.go at once :P

File: util/collate/collate_bench_test.go [hunk 1]
```diff
@@ -16,9 +16,8 @@ package collate
 
 import (
 	"math/rand"
-	"testing"
```
Comment (tisonkun) [inline util/collate/collate_bench_test.go:19]: go-imported doesn't complain on these lines? I don't like unstable format changes over and over though.

Comment (hawkingrei) [reply util/collate/collate_bench_test.go]: The change is the other pr. Now I have removed it.

**Comment by hawkingrei** [comment]
> Approve.
> 
> But I'm afraid that if you fix it partially, you may fight with CI for the failures caused by files you changed exactly.
> 
> If you're not in a hurry, I suggest you move related tests in bootstrap_test.go to upgrade_test.go at once :P

Thank you for your advice. I will distribute this task to colleagues in my group to fix them. It has created many data races in the report. I need to clear out some of it.

File: util/collate/collate_bench_test.go (hunk 1 above)
Comment (hawkingrei) [reply util/collate/collate_bench_test.go]: BTW，Can you merge https://github.com/pingcap/tidb/pull/31887 ?

**Review by hanfei1991 — APPROVED** [review]

**Pull request merged**